# In Python 3, hashlib requires a bytes-like object rather than a str.
# This commit fixes "TypeError: Strings must be encoded before hashing"
# by encoding self._url before passing it to hashlib.update().
From 577673dda5d27f48b2d6cda5cd7eba10c898e24f Mon Sep 17 00:00:00 2001
From: Dentrax <furkan.turkal@chainguard.dev>
Date: Thu, 20 Mar 2025 15:21:40 +0300
Subject: [PATCH] fix(gdapi.py): encode strings for hashing in Python 3

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>
---
 gdapi.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gdapi.py b/gdapi.py
index 21bc00a..b3ed324 100755
--- a/gdapi.py
+++ b/gdapi.py
@@ -513,9 +513,9 @@ class Client(object):
 
     def _get_schema_hash(self):
         h = hashlib.new('sha1')
-        h.update(self._url)
+        h.update(self._url.encode("utf-8"))
         if self._access_key is not None:
-            h.update(self._access_key)
+            h.update(self._access_key.encode("utf-8"))
         return h.hexdigest()
 
     def _get_cached_schema_file_name(self):
-- 
2.39.5 (Apple Git-154)

